### DISCUSSION TITLE
DNS 打不过去，给了最小配置，帮忙看一下感谢

### DISCUSSION THREAD

**Discussion by harrisliupsn** [v2fly/v2ray-core#2486] [Q&A]
需求：
本地开发的时候在浏览器简单地输入 ``app.local.host`` 和 ``app.remote.host`` 就能连上对应的 app，比如说
- 本机自建的 gitlab：``gitlab.local.host``
- 公司的 gitlab：``gitlab.remote.host``

现在已经写了一个简单的 Caddyfile，可以把 ``gitlab.local.localhost`` 映射到 ``localhost:9000``，浏览器能成功打开
有问题的一步在 v2ray 的 DNS 解析这里，一直没办法让 ``domain:local.host`` 走到定义的 DNS 规则上，
写了一个简单的测试配置，感觉在 DNS 跟 routing 的理解上什么有问题，请大佬们帮忙看一下

测试配置文件例子中
期望的行为：将 ``local.host`` 翻译成 ``127.0.0.1``，并且走 out-direct 出去
实际的行为：规则没有匹配上，一直选取的默认的 outbound，也就是 out-blocked
测试配置文件：
```json
{
  "log": {
    "level": "debug",
    "access": "access.log",
    "error": "error.log"
  },
  "inbounds": [
    {
      "tag": "in-http-dns-open",
      "listen": "0.0.0.0",
      "port": 18000,
      "protocol": "http",
      "settings": {},
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ]
      }
    }
  ],
  "outbounds": [
    {
      "tag": "out-blocked",
      "protocol": "blackhole",
      "settings": {}
    },
    {
      "tag": "out-direct",
      "protocol": "freedom",
      "settings": {
        "domainStrategy": "UseIP"
      }
    }
  ],
  "dns": {
    "hosts": {
      "local.host": "127.0.0.1"
    }
  },
  "routing": {
    "domainStrategy": "IPOnDemand",
    "rules": [
      {
        "type": "field",
        "outboundTag": "out-direct",
        "ip": [
          "127.0.0.1",
          "geoip:private"
        ]
      },
      {
        "type": "field",
        "outboundTag": "out-direct",
        "domain": [
          "local.host",
          "full:local.host",
          "domain:local.host"
        ]
      }
    ]
  }
}
```

环境：
- Windows 11 专业版 22H2 22621.1555
- V2Ray 5.3.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.20 windows/amd64)

---
Update:
稍微看了一下，我感觉是不是版本 v5 不支持 v4 的配置了？
有没有大佬能把我上面的例子改成 v5 的版本？

**Comment by nursery01** [comment]
你改hosts應該就能轉到想要的位址

我的意思是C:硬碟裡面有個hosts，你改這個瀏覽器應該可以訪問，我這邊是測試是可以

**Reply by harrisliupsn** [reply]
我试了一下，dns 部分改成 v5 的格式（staticHosts）不行，上面的最小用例在 v4 的 v2ray 也不会按预期的方式运行

**Comment by mydogshitgold** [comment]
There is a minor mistake that needs to be corrected.
In the dns.host section, "local.host" needs be changed to "domain:loca.host".

```json
"dns": {
  "hosts": {
    "domain:local.host": "127.0.0.1"
  }
}
```

Otherwise, this configuration shoulld be functional.

---

Without explicitly opening a DNS inbound and configuring your operating system to use it, your custom domain name configured in v2ray may not be recognized by many programs that perform name resolution locally.

---

Note: 
- Using `.host` may cause conflicts because it is already an established gTLD.
- The `loglevel` property is used to set the logging level in the `log` object, not the `level` property.

**Comment by harrisliupsn** [comment]
感觉 v2ray 的 routing、dns 和 outbounds 顺序确实比较迷惑，最后是实现了想要的功能，但是没有经过 v2ray
具体是利用了 mDNS 和 hosts 文件来完成的
```
# hosts
127.0.0.1 gitea.my.local
```
然后 caddy 反向代理 gitea.my.local 到想要的端口 localhost:port
